### PR TITLE
boards: stm32f412g_disco: Add arduino headers

### PR DESCRIPTION
**Pull request by KimBP**
Added arduino gpios, and the typically used aliases for
arduino_spi, arduino serial and arduino_i2c

### CODE REVIEW

File: boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			   <16 0 &gpioa 15 0>,	/* D10 */
+			   <17 0 &gpioa 7 0>,	/* D11 */
+			   <18 0 &gpioa 6 0>,	/* D12 */
+			   <19 0 &gpioa 5 0>,	/* D13 */
+			   <20 0 &gpiob 9 0>,	/* D14 */
+			   <21 0 &gpiob 10 0>;	/* D15 */
+	};
+};
+
+arduino_i2c: &i2c2 {
```
Comment (ABOSTM) [inline boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi:38]: Generally, only alias is in arduino_r3_connector.dtsi, the i2c, spi and usart should be defined in stm32f412g_disco.dts.
Also, missing 	`status = "okay";` for i2c, spi and usart.

supported of arduino_gpio, arduino_i2c and arduino_spi, can be added to nucleo_l476rg.yaml
(see nucleo_l476rg as example)

**Review by ABOSTM — CHANGES_REQUESTED** [review]
@KimBP, thanks for your contribution.

File: boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi (hunk 1 above)
Comment (KimBP) [reply boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi:38]: I deliberately left out the Okay part. There might be situations where you don't want to activate e.g. the SPI pins but use an alternate function.
Regarding where to place i2c, spi and usart: 
This MCU has several pins you can use for e.g. spi1. Aliasing arduino_spi to spi1 is only valid with exactly this pin configuration even though the aliasing is probably independent of the pin settings.
I'll have a look at nucleo_l476rg

Comment (erwango) [reply boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi:38]: > There might be situations where you don't want to activate e.g. the SPI pins but use an alternate function.

What we intend to provide here is a default board configuration.
cf https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations

Comment (KimBP) [reply boards/arm/stm32f412g_disco/arduino_r3_connector.dtsi:38]: Yeah I see that. Will fix it

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Pull request merged**